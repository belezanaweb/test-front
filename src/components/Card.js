import React, { Fragment } from 'react'; 
import styled from 'styled-components'; 

const Title = styled.div`
    height: 17px;
    width: 319px;
    color: #999;
    font-size: 14px;
    font-weight: 700;
    line-height: 17px;
    padding-top: 2vh;
    padding-bottom: 2vh;
`

const Box = styled.div`
    width: 341px;
    border-radius: 3px;
    background-color: #FFF;
    box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
`

const Card = (props) => {
    return (
        <Fragment>
            <Title>
                {props.title}
            </Title>
            <Box>
                {props.children}
            </Box> 
        </Fragment>

    )
}

export default Card;