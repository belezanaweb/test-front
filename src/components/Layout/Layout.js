import styled from 'styled-components'

export const Layout = styled.main`
  display: flex;
  flex-direction: column;
`

export const LayoutGrid = styled.div`
  flex-grow: 1;
  display: grid;
  grid-gap: 20px;
  padding: 0 9px 20px;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;

  @media only screen and (min-width: 768px) {
    margin-top: 40px;
    grid-template-columns: 1fr 340px;
  }
`

export const LayoutContainer = styled.div`
  display: grid;
  align-content: start;
  grid-gap: 20px;
`

export const LayoutAside = styled.aside`
  display: grid;
  align-content: start;
  grid-gap: 20px;

  @media only screen and (min-width: 768px) {
    margin-top: 20px;
  }
`
