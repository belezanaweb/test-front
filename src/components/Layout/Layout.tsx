import React from 'react';
import styled from 'styled-components';

import Steps, { StyledSteps, StyledStep } from '../Steps';

const Container = styled.div`
  padding: 0 10px 20px;
`;

const StyledLayout = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 768px) {
    flex-direction: row;
    margin: 50px auto 0;
    max-width: 768px;
    width: 100%;

    ${Container} {
      width: 512px;
    }

    ${StyledSteps} {
      background: transparent;
      box-shadow: none;
      height: 100%;
      margin-top: 120px;
      min-width: 256px;
      
      ol {
        align-items: flex-end;
        flex-direction: column;
      }
    }

    ${StyledStep} {
      height: 40px;
      margin-right: 20px;
    }
  }
`;

const Layout: React.FC = ({ children }) => {
  return (
    <StyledLayout>
      <Steps />
      <Container>
        {children}
      </Container>
    </StyledLayout>
  );
};

export default Layout;