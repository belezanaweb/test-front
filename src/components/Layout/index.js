export { default } from './Layout'
export { Container } from './styles'
