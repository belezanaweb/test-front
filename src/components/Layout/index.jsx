import React from 'react'

import Header from '../Header'
import Footer from '../Footer'

function Layout({ children }) {
  return (
    <main>
      <Header />
      { children }
      <Footer />
    </main>
  )
}
export default Layout

