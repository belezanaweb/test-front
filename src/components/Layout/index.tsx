import { Flex } from '@chakra-ui/react'
import Navbar from '../Navbar'
import { Outlet } from 'react-router-dom'

export const Layout = () => (
  <Flex flexDir={'column'} h="100vh">
    <Navbar />
    <Flex
      h="calc(100vh - 40px)"
      gap={'20px'}
      flexDir={['column', 'row']}
      justifyContent={['space-between', 'center']}
      align={'flex-start'}
    >
      <Outlet />
    </Flex>
  </Flex>
)
