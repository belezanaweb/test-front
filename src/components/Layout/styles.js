import styled from 'styled-components'
import { device } from '../../styles/breakpoints'

export const Container = styled.div`
  width: 100%;
  height: 100%;
  padding: 0 var(--spacing-2);
  margin: 0 auto;
  max-width: 600px;

  @media ${device.desktop} {
    max-width: 900px;
  }
`

export const Layout = styled.div`
  display: grid;
  min-height: 100vh;
  grid-template-rows: var(--spacing-8) 1fr;
`
