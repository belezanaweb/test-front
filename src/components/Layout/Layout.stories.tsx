import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import theme, { Theming } from '../../theme';

import Card from '../Card';
import Text from '../Text';

import Layout from './Layout';

const Provider = (storyFn: () => React.ReactNode) => (
  <Theming theme={theme}>
    <BrowserRouter>
      {storyFn()}
    </BrowserRouter>
  </Theming>
);

export default {
  component: Layout,
  decorators: [Provider],
  title: 'Layout',
};

export const Default = () => (
  <Layout>
    <Card css={{
      alignItems: 'center',
      marginTop: '30px',
      height: '360px',
      justifyContent: 'center',
    }}>
      <Text>Resize Window</Text>
    </Card>
  </Layout>
);
