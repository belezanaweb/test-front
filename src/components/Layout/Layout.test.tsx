import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { provider } from '../../utils/testUtils';

import Card from '../Card';
import Text from '../Text';

import Layout from './Layout';

describe('<Layout />', () => {
  it('should render Layout', () => {
    const { container } = provider((
      <BrowserRouter>
        <Layout>
          <Card css={{
            alignItems: 'center',
            marginTop: '30px',
            height: '360px',
            justifyContent: 'center',
          }}>
            <Text>Resize Window</Text>
          </Card>
        </Layout>
      </BrowserRouter>
    ), render);

    expect(container).toMatchSnapshot();
  });
});
