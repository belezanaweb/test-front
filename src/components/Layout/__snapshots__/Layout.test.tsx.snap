// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render Layout 1`] = `
<div>
  <div
    class="sc-AxmLO kqgTeu"
  >
    <nav
      class="sc-AxgMl eSvPIa"
    >
      <ol>
        <li
          class="sc-AxheI hsfhcz"
        >
          <a>
            Sacola
          </a>
        </li>
        <li
          class="sc-AxheI hsfhcz"
        >
          <a>
            Pagamento
          </a>
        </li>
        <li
          class="sc-AxheI hsfhcz"
        >
          <a>
            Confirmação
          </a>
        </li>
      </ol>
    </nav>
    <div
      class="sc-Axmtr gWwcVn"
    >
      <div
        class="sc-AxiKw sc-AxhCb jPYptu"
      >
        <span
          class="sc-AxhUy aBiJA"
        >
          Resize Window
        </span>
      </div>
    </div>
  </div>
</div>
`;
